lms/models/region.py

```diff
@@ -56,22 +52,19 @@ def get_region(cls) -> Region:
         return cls._current_region
 
     @classmethod
-    def set_region(cls, authority: str):
+    def set_region(cls, authority: str, code: str):
```

Comment:
`set_region()` just calls `from_code()` and caches the result. Had to add the `code` argument to it now that it can no longer look up the region in a hard-coded mapping of `authority`'s to regions